web/src/main/kotlin/no/nav/su/se/bakover/web/services/personhendelser/PersonhendelseMapper.kt

```diff
@@ -0,0 +1,95 @@
+        message: ConsumerRecord<String, EksternPersonhendelse>,
+    ): Either<HendelseMapperException, Personhendelse.Ny> {
+        val key: String = message.key()
+        val offset: Long = message.offset()
+        val personhendelse: EksternPersonhendelse = message.value()
+
+        val aktørId = hentAktørId(personhendelse, key).getOrHandle {
+            return it.left()
+        }
+
+        return when (personhendelse.getOpplysningstype()) {
+            Opplysningstype.DØDSFALL.value -> {
+                val dødsdato = personhendelse.getDoedsfall().get().let {
+                    if (it.getDoedsdato().isPresent) it.getDoedsdato().get() else null
+                }
+
+                Personhendelse.Ny(
+                    hendelseId = personhendelse.getHendelseId(),
+                    offset = offset,
+                    gjeldendeAktørId = aktørId,
+                    endringstype = hentEndringstype(personhendelse.getEndringstype()),
+                    personidenter = personhendelse.getPersonidenter(),
+                    hendelse = Personhendelse.Hendelse.Dødsfall(dødsdato = dødsdato),
+                ).right()
+            }
+
+            Opplysningstype.UTFLYTTING_FRA_NORGE.value -> {
+                val utflyttetDato = personhendelse.getUtflyttingFraNorge().get().let {
+                    if (it.getUtflyttingsdato().isPresent) it.getUtflyttingsdato().get() else null
+                }
+
+                Personhendelse.Ny(
+                    hendelseId = personhendelse.getHendelseId(),
+                    offset = offset,
+                    gjeldendeAktørId = aktørId,
+                    endringstype = hentEndringstype(personhendelse.getEndringstype()),
+                    personidenter = personhendelse.getPersonidenter(),
+                    hendelse = Personhendelse.Hendelse.UtflyttingFraNorge(utflyttingsdato = utflyttetDato),
+                ).right()
+            }
+            else -> {
+                IkkeAktuellOpplysningstype.left()
+            }
+        }
+    }
+
+    private fun hentAktørId(
+        personhendelse: EksternPersonhendelse,
+        key: String,
+    ): Either<HendelseMapperException.KunneIkkeHenteAktørId, AktørId> {
+        val id = key.substring(6) // Nyckeln på Kafka-meldinger fra Pdl er aktør-id prepend:et med 6 rare tegn.
+        val idFinnesSomPersonident = personhendelse.getPersonidenter().any { it == id }
+
+        if (!idFinnesSomPersonident) {
+            return HendelseMapperException.KunneIkkeHenteAktørId.left()
+        }
+
+        return AktørId(id).right()
+    }
+
+    private fun hentEndringstype(endringstype: Endringstype) =
+        when (endringstype) {
+            Endringstype.OPPRETTET -> Personhendelse.Endringstype.OPPRETTET
+            Endringstype.KORRIGERT -> Personhendelse.Endringstype.KORRIGERT
+            Endringstype.ANNULLERT -> Personhendelse.Endringstype.ANNULLERT
+            Endringstype.OPPHOERT -> Personhendelse.Endringstype.OPPHOERT
```

Comment:
Lurer litt på hvordan vi skal bruke disse i praksis, gitt at vi egentlig bare lager en Gosys-oppgave når vi fanger opp en relevant hendelse.. hva gjør vi da hvis vi en uke senere fanger opp en korreksjon eller annullering? 

Antar jo egentlig at det ikke er så relevant, siden saksbehandleren som plukker opp den originale Gosys-oppgaven antageligvis uansett sjekker om det stemmer at vedkommende er død eller utflyttet. Hvis de uansett gjennomfører revurderingen blir det jo litt sent å si "kødda!" i etterkant. Men må kanskje sjekke litt med fag hva de tenker her, men det trengs jo ikke for nå.

Reply:
Jeg foreslår at vi bare legger ved informasjonen i Gosys oppgaven, og dersom det kommer en ny hendelse som korrigerer/annulerer lager vi bare enda en Gosys-oppgave? Så kan saksbehandlerene bare kommentere og ferdigstille den direkte?

Reply:
Ja, høres greit ut det. Så kan vi ta en diskusjon med fag når det skjer i praksis.